### PR TITLE
Fix some xcode 7 (llvm) warnings

### PR DESCRIPTION
**Pull request by martinlindhe**
Also tested compiling under Ubuntu with these changes, couldn't spot any issues.

### CODE REVIEW

File: src/video_core/rasterizer.cpp [hunk 1]
```diff
@@ -967,6 +967,8 @@ static void ProcessTriangleInternal(const Shader::OutputVertex& v0,
                         UNIMPLEMENTED();
                         break;
                     }
+
+                    return NULL;
```
Comment (yuriks) [inline src/video_core/rasterizer.cpp]: The value being returned isn't a pointer, so this makes no sense. Return a zeroed vector instead or something.

File: src/video_core/rasterizer.cpp [hunk 2]
```diff
@@ -1000,6 +1002,8 @@ static void ProcessTriangleInternal(const Shader::OutputVertex& v0,
...
```
Comment (yuriks) [inline src/video_core/rasterizer.cpp]: Likewise.

**Comment by martinlindhe** [comment]
@yuriks: thanks, i have updated the patch with your feedback

File: src/video_core/rasterizer.cpp [hunk 3]
```diff
@@ -967,6 +967,8 @@ static void ProcessTriangleInternal(const Shader::OutputVertex& v0,
                         UNIMPLEMENTED();
                         break;
                     }
+
+                    return 0;
```
Comment (bunnei) [inline src/video_core/rasterizer.cpp]: This should return `Math::Vec3<u8>(0, 0, 0);`

Comment (bunnei) [inline src/video_core/rasterizer.cpp]: Actually, just `{}` should be fine, I think

**Comment by bunnei** [comment]
LGTM after my one comment is addressed. Please squash down to a single commit.

**Comment by martinlindhe** [comment]
@bunnei: updated the patch from your feedback and squashed into a single commit

**Comment by bunnei** [comment]
@martinlindhe did you screw up the rebase? You have some of @lioncash's commits in there now...

**Comment by martinlindhe** [comment]
@bunnei: jeez, sorry for the trouble. should be ok now

**Comment by bunnei** [comment]
LGTM

**Pull request merged**